lib/Elevate/OS.pm

```diff
@@ -0,0 +1,133 @@
+        die "This script is only designed to upgrade the following OSs:\n\n$supported_distros\n";
+    }
+
+    return $OS;
+}
+
+## NOTE: private methods (beginning with _) are NOT allowed in this list!
+my %methods;
+
+BEGIN {
+
+    # The key specifies what the method that all platforms we support.
+    # The value specifies how many args the method is designed to take.
+    %methods = map { $_ => 0 } (
+        ### General distro specific methods.
+        'available_upgrade_paths',        # This returns a list of possible upgrade paths for the OS
+        'default_upgrade_to',             # This is the default OS that the current OS should upgrade to (i.e. CL7->CL8, C7->A8)
+        'disable_mysql_yum_repos',        # This is a list of mysql repo files to disable
+        'ea_alias',                       # This is the value for the --target-os flag used when backing up an EA4 profile
+        'elevate_rpm_url',                # This is the URL used to install the leapp RPM/repo
+        'is_supported',                   # This is used to determine if the OS is supported or not
+        'leapp_can_handle_epel',          # This is used to determine if we can skip removing the EPEL repo pre_leapp or not
+        'leapp_can_handle_imunify',       # This is used to determine if we can skip the Imunify component or not
+        'leapp_can_handle_kernelcare',    # This is used to determine if we can skip the kernelcare component or not
+        'leapp_can_handle_python36',      # This is used to determine if we can skip the python36 blocker or not
+        'leapp_data_pkg',                 # This is used to determine which leapp data package to install
+        'leapp_flag',                     # This is used to determine if we need to pass any flags to the leapp script or not
+        'name',                           # This is the name of the OS we are upgrading from (i.e. CentOS7, or CloudLinux7)
+        'pretty_name',                    # This is the pretty name of the OS we are upgrading from (i.e. 'CentOS 7')
+        'vetted_mysql_yum_repo_ids',      # This is a list of known mysql yum repo ids
+        'vetted_yum_repo',                # This is a list of known yum repos that we do not block on
+    );
+}
+
+sub supported_methods {
+    return sort keys %methods;
+}
+
+our $AUTOLOAD;
+
+sub AUTOLOAD {
+    my $sub = $AUTOLOAD;
+    $sub =~ s/.*:://;
+
+    exists $methods{$sub} or Carp::Croak("$sub is not a supported data variable for Elevate::OS");
+
+    my $i   = instance();
+    my $can = $i->can($sub) or Carp::Croak( ref($i) . " does not implement $sub" );
+    return $can->( $i, @_ );
+}
+
+sub DESTROY { }    # This is a must for autoload modules
+
+=head1 can_upgrade_to
+
+This returns true or false depending on whether the current OS
+is able to upgrade to the requested OS or not.
+
+=cut
+
+sub can_upgrade_to ($flavor) {
+    return grep { $_ eq $flavor } Elevate::OS::available_upgrade_paths();
```

Comment:
We resisted doing boolean checks like this in Cpanel::OS. I get why this makes checks easy. 🤷🏻 Not blocking over this just grumbling.